### PR TITLE
Add benchmark for roberta prepoc pipelines

### PR DESCRIPTION
**Pull request by parmeet**
## Summary 
This PR adds benchmark for end-2-end preproc pipeline for RoBERTa based text pre-processing. All the pipelines are based on DataPipes.
* **Pipeline 1 (benchmark_roberta_datapipe):** Using operators/transforms implemented in torchtext and are directly applied on text
* **Pipeline 2 (benchmark_roberta_dataframe):** This pipeline first convert the input text (python tuples) into DataFrame of sizes batch_size on the fly. It then does text processing directly on dataframe.
  * **Pipeline 2.2 (Using Ops as UDFs):** This pipeline uses operators implemented in torchtext and are applied to TA dataframe as a UDFs using it's transform API  
  * **Pipeline 2.1 (Using Native Ops):** This pipeline used native operators implemented directly in TorchArrow 

## Results
Current end-2-end benchmarking results:
Settings: Macbook pro, SST2 dataset, batch_size 32, single run

Pipeline 1: ~ 5 seconds
Pipeline 2.1: ~28 seconds
Pipeline 2.2: ~73 seconds

### CODE REVIEW

File: benchmark/benchmark_roberta_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        tokens = self.add_eos(tokens)
+        input["tokens"] = tokens
+        return input
+
+
+def batch_tokenize(input: List[str], tokenizer: Callable) -> List[List[str]]:
+    return tokenizer(input)
+
+
+class RobertaTransformDataFrame(Module):
```
Comment (Nayef211) [inline benchmark/benchmark_roberta_pipeline.py:51]: Same question as above for `RobertaTransformDataFrame`

File: benchmark/benchmark_roberta_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+        # Apply transformation on DataFrame
+        train_dp = train_dp.map(transform)
+
+        # Remove not required columns
+        train_dp = train_dp.map(lambda x: x.drop(["text"]))
+
+        # convert DataFrame to tensor (This will yeild named tuple)
+        train_dp = train_dp.map(lambda x: x.to_tensor({"tokens": tap.PadSequence(padding_value=1)}))
+
```
[the remark this answers is not in the document]
Comment (parmeet) [reply benchmark/benchmark_roberta_pipeline.py:127]: hmm,  not really. I think the only common code is where we map the transform, right?

File: benchmark/benchmark_roberta_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+import torchtext.functional as F
+import torchtext.transforms as T
+from benchmark.utils import Timer
+from torch.hub import load_state_dict_from_url
+from torch.nn import Module
+from torch.utils.data import DataLoader
+from torchtext.datasets import DATASETS
+
+
+class RobertaTransformDataPipe(Module):
```
[the remark this answers is not in the document]
Comment (parmeet) [reply benchmark/benchmark_roberta_pipeline.py:16]: I think it's a good point. Let me see if there is an elegant way to reuse the code.

File: benchmark/benchmark_roberta_pipeline.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nayef211) [reply benchmark/benchmark_roberta_pipeline.py:127]: Actually taking another look the similarities are minor and I guess there's not really a point in overcomplicating the simple benchmarking pipeline. Leaving this as is sounds good to me!

File: benchmark/benchmark_roberta_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        self.add_bos = T.AddToken(token=0, begin=True)
+
+        # Add EOS token to the end of sentence
+        self.add_eos = T.AddToken(token=2, begin=False)
+
+    def forward(self, input: ta.DataFrame) -> ta.DataFrame:
+        input["tokens"] = ta_F.bpe_tokenize(self.tokenizer, input["text"])
+        input["tokens"] = input["tokens"].list.slice(stop=254)
+        token_ids = ta_F.lookup_indices(self.vocab, input["tokens"])
+        token_ids._dtype = dt.List(dt.int64)
```
Comment (wenleix) [inline benchmark/benchmark_roberta_pipeline.py:100]: this hack should be no longer needed once https://github.com/facebookresearch/torcharrow/pull/361 merged.

Comment (parmeet) [reply benchmark/benchmark_roberta_pipeline.py:100]: Sounds good! Thanks @wenleix for the fix :)

Comment (erip) [reply benchmark/benchmark_roberta_pipeline.py:100]: Note that torcharrow doesn't support even "relatively old" hardware (i.e., https://github.com/facebookresearch/torcharrow/issues/161) which may leak make these features unavailable to OS X users. If this becomes a hard dep on torchtext, some users may not even be able to use torchtext anymore (me included!)

Comment (wenleix) [reply benchmark/benchmark_roberta_pipeline.py:100]: @erip : Thanks for raising the concern. Could you try it again? -- we should have recently fixed it issue and TorchArrow should be able to run on SSE-only machines (e.g. https://github.com/facebookresearch/torcharrow/pull/315)

File: benchmark/benchmark_roberta_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        # TODO: Figure out how to create DataFrame of larger size and create smaller batches
+        train_dp = train_dp.dataframe(columns=columns, dataframe_size=batch_size)
+
+        # Apply transformation on DataFrame
+        train_dp = train_dp.map(transform)
+
+        # Remove not required columns
+        train_dp = train_dp.map(lambda x: x.drop(["text"]))
+
+        # convert DataFrame to tensor (This will yeild named tuple)
```
Comment (Nayef211) [inline benchmark/benchmark_roberta_pipeline.py:73]: nit: "yield"

**Review by Nayef211 — APPROVED** [review]
LGTM! Thanks @parmeet for updating the benchmarks to reuse code from our previously implemented examples. These benchmarks will serve as a great starting point to understanding where the performance bottlenecks might be for our torcharrow integration work.

**Pull request merged**